### PR TITLE
Switch to PSR-4

### PR DESCRIPTION
**Pull request by afurculita**
PSR-4 offers a much cleaner directory structure.

All the files from src/JMS/Serializer were moved to the root of the src/ directory. Same with the files from tests/Jms/Serializer.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,11 +1,12 @@
 language: php
 
+sudo: false
+
 matrix:
   allow_failures:
     - php: hhvm
 
 php:
-  - 5.3
```
Comment (Soullivaneuh) [inline .travis.yml]: This should not be removed.

AFAIK, php 5.3 is still supported by this library and this is not related to your PR.

File: .travis.yml [hunk 2]
```diff
@@ -1,11 +1,12 @@
 language: php
 
+sudo: false
```
Comment (Soullivaneuh) [inline .travis.yml]: Not related to the PR.

File: .travis.yml (hunk 1 above)
Comment (afurculita) [inline .travis.yml]: As of version 1.0, this library does not support php 5.3. Proof: composer.json is set to require php >=5.4 and code like https://github.com/schmittjoh/serializer/blob/1.0.0/src/JMS/Serializer/Serializer.php#L82 is not supported in 5.3.

I know it is not related to my PR, but it broke a lot of tests

File: .travis.yml (hunk 2 above)
Comment (afurculita) [inline .travis.yml]: I thought this doesn't need its own PR. Should I create a new PR for it?

Comment (Soullivaneuh) [inline .travis.yml]: I'm not the owner of this repo but IMHO yes, it should be on another PR.

**Comment by goetas** [comment]
@stof In your opinion, switching to PSR4 should be considered a BC break?

**Comment by goetas** [comment]
Moving to PSR4 can be done later. 
Merging this will cause a lot of conflicts with currently opened PR
